docs/private-registries.md

```diff
@@ -23,19 +23,23 @@ password `auth` string:
 ```
 
 `<REGISTRY_NAME>` needs to be replaced by the name of your private registry
-(e.g., `my-private-registry.example.org`)
-
-!!! important "Using private images on docker hub"
-    When using private images on docker hub, the containers beeing watched needs to use the full image name, including the repository prefix `index.docker.io`.
-    So instead of
-    ```
-    docker run -d myuser/myimage
-    ```
-    you would run it as
-    ```
-    docker run -d index.docker.io/myuser/myimage
-    ```
-
+(e.g., `my-private-registry.example.org`).
+
+!!! info "Using private images on Docker Hub"
+    To access private repositories on Docker Hub,
+    `<REGISTRY_NAME>` should be `index.docker.io`.
```

Comment:
```suggestion
    `<REGISTRY_NAME>` should be `docker.io`.
```
If using `docker.io` is fully supported, we might as well put it in the docs for new users, no?

Reply:
Agreed, but this "full support" is not what it seems. Docker uses `docker.io` internally as the default registry domain, but still expects `config.json` to contain credentials for `index.docker.io`. Worse, it does not recognize any other key than `https://index.docker.io/v1/`, which means running `docker pull IMAGE` fails with a config file that currently works for Watchtower.

Reply:
I have amended my last 2 commits to mimic this behavior in Watchtower and reflect it in the docs. For backwards compatibility, Watchtower will still recognize the `index.docker.io` credential entry, but also try `https://index.docker.io/v1/` if that doesn't exist.

Reply:
I have amended my last 2 commits to "comply" with Docker's own behavior and reflect this in the documentation.

Reply:
Note: the CLI [treats `index.docker.io` as an exception](https://github.com/docker/cli/blob/65438e008c20a6125a1319eca07dcc3d7d4e38eb/cli/command/registry.go#L73-L76) to the rule of how credentials are stored in `config.json`, and [a comment in `docker/cli`](https://github.com/docker/cli/blob/850fb6921cf7a208fd47d83074f27d0eb7ae268e/cli/config/credentials/file_store.go#L36-L42) remarks that this is legacy.

Would you agree that we should play it safe and ensure that a `config.json` manually generated according to the Watchtower docs should also work with the Docker CLI, even though this means we tell users to go along with this legacy, inconsistent behavior?